src/relayclient/RelayClient.ts

```diff
@@ -80,8 +81,34 @@ export class RelayClient {
     configOverride: Partial<GSNConfig>,
     overrideDependencies?: Partial<GSNDependencies>
   ) {
-    this.config = configureGSN(configOverride)
-    const dependencies = getDependencies(this.config, provider, overrideDependencies)
+    this.partialConfig = { provider, configOverride, overrideDependencies }
+    this.logger = console
+  }
+
+  initializing?: Promise<void>
+
+  async init (): Promise<this> {
```

Comment:
No, wait, we can't have both `init` and `_init`, this will cause a ton of confusion. Why do they have to be two different methods?

Reply:
Init is api to call, and does the sync check.
_init is internal. I'll rename it to _initInternal
I used to use the runExclusive mechanism, but it trashes any stack trace of errors during init.